### PR TITLE
Added collaboration message to all sulu core-bundles

### LINKED ISSUE #2045: Showing current collaborators of snippets
The same mechanism as in pages should be implemented in the snippets

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2045 
| Related issues/PRs | none
| License | MIT
| Documentation PR | https://github.com/sulu-io/sulu-docs/pull/194

#### What's in this PR?

This PR add the collaborator message to the snippet-form. Additionally it extracts the initialization of the component into an extension which can be used in every component.

#### Why?

The extension should enabled the developer to easily integrate the collaboration message into their own form.

The question now is should we integrate it also in all sulu forms? /cc @chirimoya @berndhep 

#### Example Usage

__With auto-start:__
~~~php
return {
...
    collaboration: function() {
        return {
            id: this.options.id,
            type: 'snippet'
        };
    }
...
}
~~~

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js [hunk 1]
```diff
@@ -28,6 +28,15 @@ define(['app-config', 'config', 'sulucontent/components/content/preview/main'],
             };
         },
 
+        collaboration: function() {
+            return {
+                autoStart: false,
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: Why do we set `autoStart` here to false? Actually we could get rid of this setting, or is there any use case for it? If not necessary I would always assign an object to `collaboration` and just start it. From what I see [here](https://github.com/sulu/sulu/pull/2201/files#diff-c124d8cf9d9f11172e744e9e64c33739R196) it is started anyway if the collaboration component is registered, so I don't see a point for this flag.

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: at least in the content form you have started the component quite late in the first version. i have tested it also without autoStart but the component was remove right after creating it. therefor i have introduced this flag for the content form.

**Comment by wachterjohannes** [comment]
@danrot the collaboration messages are now integrated in all sulu forms

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js [hunk 2]
```diff
@@ -20,6 +20,14 @@ define([
     var defaults = {};
 
     return {
+
+        collaboration: function() {
+            return {
+                id: this.options.id,
+                type: 'collection'
+            };
+        },
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js]: Do we really also want that in the entire collection? Could imagine that collections are often opened at the same time, would only be a problem for the overlays. I think I would just skip this component for now...

**Comment by danrot** [comment]
Why is the if statement for `this.options.data` only required for some components?

**Comment by danrot** [comment]
When thinking of it this check for `this.options.data` should be included in every form using the collaboration, since it is not possible edit what another user is currently editing :see_no_evil:

**Pull request merged**